keg/yaml_utils.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2020 SUSE Software Solutions Germany GmbH. All rights reserved.
+#
+# This file is part of keg.
+#
+# keg is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# keg is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with keg. If not, see <http://www.gnu.org/licenses/>
+#
+
+from glob import glob
+from pathlib import Path
+import os
+import yaml
+
+
+def _get_versioned_source_files(src_path, include_paths):
```

Comment:
Should be recursive or use os.walk() to eliminate explicit tracking of where we have been

Reply:
The tracking of where we have been is to avoid importing the same file twice at a different (and unwanted) point. For instance, SLE15-SP2 image definitions uses include paths `sle15-sp2` and `sle15-sp1`. Besides others, the following directories are scanned for yaml files:
```
data/csp/ec2
data/csp/ec2/sle15
data/csp/ec2/sle15/sp1
data/csp/ec2/sle15
data/csp/ec2/sle15/sp2
```
The order how the files are added to the list is important as the most specific ones need to be merged last. However, since `data/csp/ec2/sle15` is twice in the list, so anything included from `data/csp/ec2/sle15/sp1` could potentially be overwritten. Hence the explicit check.

I've tried to make it somewhat more elegant and also looked into using os.walk() but couldn't make it work.